test/foils.test.js

```diff
@@ -0,0 +1,81 @@
+const { describe, it, before } = global;
+const assert = require("assert");
+const jsdom = require("jsdom");
+/* this weird stuff basically makes jsPsych testable in Node */
+const { JSDOM } = jsdom;
+const { window } = new JSDOM("<!DOCTYPE html><p>Hello world</p>");
+global.window = window;
+global.document = window.document;
+
+const { populateExemplars} = require("../src/exemplars");
+const { createFoils} = require("../src/foils");
+
+const paramSimple = {
+  exemplarTypes: ['NNN', 'BBB', 'NNB', 'NBB'],
+  numExemplarsPerType: 1,
+  foilTestedOn: [1, 0, 0, 1],
+  foilTestedType: [false, true, false, true]
+}
+
+const paramHalfRand = {
+  exemplarTypes: ['NNN', 'rand_BBB', 'NNB', 'rand_NBB'],
+  numExemplarsPerType: 1,
+  foilTestedOn: [1, 0, 0, 1],
+  foilTestedType: [false, true, false, true]
+}
+
+describe("Foil from simple params", () => {
+  let exemplars, foils;
+  before(() => {
+    exemplars = populateExemplars(paramSimple);
+    foils = createFoils(exemplars, paramSimple);
+    randExemplars = populateExemplars(paramHalfRand)
+    randFoils = createFoils(randExemplars, paramHalfRand);
+  });
+
+  it("created the correct number of foils", () => {
+    assert.equal(
+      Object.keys(exemplars).length,
+      Object.keys(foils).length
+    );
+  });
+
+  it("is not the same as exemplars", () => {
+    assert.notDeepEqual(
+      exemplars,
+      foils
+    );
+  });
+  it("Creates unique foils", () => {
```

Comment:
I'm being annoying here but could you make the descriptions all begin with lowercase characters? Just to keep the formatting consistent

Reply:
Got it! I also cleaned up more of the wording to make more sense